src/main.rs

```diff
@@ -125,242 +125,307 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
-        new_score_upper_layout.insert_row(0);
-        new_score_upper_layout.insert_row(1);
-        new_score_upper_layout.insert_row(2);
-        new_score_upper_layout.insert_row(3);
-        new_score_upper_layout.insert_row(4);
-
-        new_score_upper_layout.set_column_spacing(BUTTON_SPACING.try_into().unwrap());
-        new_score_upper_layout.set_row_spacing(BUTTON_SPACING.try_into().unwrap());
+        new_score_layout.set_column_spacing(BUTTON_SPACING.try_into().unwrap());
+        new_score_layout.set_row_spacing(BUTTON_SPACING.try_into().unwrap());
 
         let score_white_select = new_button("WHITE", "white", None);
         let score_black_select = new_button("BLACK", "black", None);
-        let score_cancel = new_button("CANCEL", "red", Some((-1, BUTTON_STANDARD_HEIGHT)));
-        let score_submit = new_button("SUBMIT", "green", Some((-1, BUTTON_STANDARD_HEIGHT)));
-
+        let score_cancel = new_button("CANCEL", "red", None);
+        let score_submit = new_button("SUBMIT", "green", None);
+        /*
+                let score_white_timeout = new_button("WHITE TIMEOUT", "white", None);
+                let score_referee_timeout = new_button("REFEREE TIMEOUT", "yellow", None);
+                let score_black_timeout = new_button("BLACK TIMEOUT", "black", None);
+        */
         let score_player_number = gtk::Label::new(Some("#P"));
         score_player_number.get_style_context().add_class("gray");
-        score_player_number.set_size_request(-1, LABEL_STANDARD_HEIGHT);
 
         let score_keypad = new_keypad();
 
-        new_score_upper_layout.attach(&score_keypad, 0, 0, 1, 4);
-        new_score_upper_layout.attach(&score_white_select, 1, 0, 1, 2);
-        new_score_upper_layout.attach(&score_black_select, 1, 2, 1, 2);
-        new_score_upper_layout.attach(&score_cancel, 0, 4, 1, 1);
-        new_score_upper_layout.attach(&score_player_number, 1, 4, 1, 1);
-        new_score_upper_layout.attach(&score_submit, 2, 4, 1, 1);
-
-        let timeout_ribbon = new_timeout_ribbon();
-
-        new_score_layout.pack_start(&new_score_upper_layout, false, false, 0);
-        new_score_layout.pack_start(&timeout_ribbon, false, false, 0);
+        let new_score_timeout_ribbon = new_timeout_ribbon();
+
+        new_score_layout.attach(&score_keypad, 0, 0, 4, 7);
+        new_score_layout.attach(&score_white_select, 4, 0, 4, 3);
+        new_score_layout.attach(&score_black_select, 8, 0, 4, 3);
+        new_score_layout.attach(&score_cancel, 0, 7, 4, 2);
+        new_score_layout.attach(&score_player_number, 4, 7, 4, 2);
+        new_score_layout.attach(&score_submit, 8, 7, 4, 2);
+        new_score_layout.attach(&new_score_timeout_ribbon, 0, 9, 12, 2);
+        /*
+                new_score_layout.attach(&score_white_timeout, 0, 9, 3, 2);
+                new_score_layout.attach(&score_referee_timeout, 3, 9, 6, 2);
+                new_score_layout.attach(&score_black_timeout, 9, 9, 3, 2);
+        */
```

Comment:
Remove

Reply:
yeah, I left it in because that is what I had before I replaced it with the timeout ribbon, I wasn't sure which you would prefer in the long run

Reply:
went with the grid layout timeout buttons so kept this code and removed the timeout ribbons